sdk/digitaltwins/test-resources.bicep

```diff
@@ -0,0 +1,109 @@
+param testApplicationOid string {
+    metadata: {
+        description: 'The client OID to grant access to test resources.'
+    }
+}
+
+param baseName string {
+    default : resourceGroup().name
+    minLength: 6
+    maxLength: 50
+    metadata: {
+        description: 'The base resource name.'
+    }
+}
+
+param location string {
+    default: resourceGroup().location
+    metadata: {
+        description: 'The location of the resource. By default, this is the same as the resource group.'
+    }
+} 
+
+var roleDefinitionId = '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/8e3af657-a8ff-443c-a75c-2fe8c4bcb635'
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2018-09-01-preview' = {
+    name: guid(resourceGroup().id)
+    properties: {
+        roleDefinitionId: roleDefinitionId
+        principalId: testApplicationOid
+    }
+}
+
+resource digitaltwin 'Microsoft.DigitalTwins/digitalTwinsInstances@2020-03-01-preview' = {
+    name: baseName
+    location: location
+    sku: {
+        name: 'S1'
+    }
+    properties: {}
+}
+
+resource digitaltwinRoleAssignment 'Microsoft.DigitalTwins/digitalTwinsInstances/providers/roleAssignments@2020-03-01-preview' = {
+    name: '${digitaltwin.name}/Microsoft.Authorization/${guid(uniqueString(baseName))}'
+    properties: {
+        roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/bcd981a7-7f74-457b-83e1-cceb9e632ffe'
+        principalId: testApplicationOid
+    }
+}
+
+resource eventHubNamespace 'Microsoft.EventHub/namespaces@2018-01-01-preview' = {
+    name: baseName
+    location: location
+    sku: {
+        name: 'Standard'
+        tier: 'Standard'
+        capacity: 1
+    }
+    properties: {
+        zoneRedundant: false
+        isAutoInflateEnabled: false
+        maximumThroughputUnits: 0
+        kafkaEnabled: true
+    }
+}
+
+resource eventHubNamespaceEventHub 'Microsoft.EventHub/namespaces/eventhubs@2017-04-01' = {
+    name: '${eventHubNamespace.name}/${baseName}'
```

Comment:
Super cool string interpolation style concatenation instead of the long [concat()]